lib/facility.rb

```diff
@@ -1,14 +1,76 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def has_service?(service)
+    @services.include?(service)
+   end
+
+  def allow_service(service)
+    if has_service?(service)
+      yield
+    else
+      false
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if 
+      @registered_vehicles << vehicle
+      vehicle.set_date
+      vehicle.give_plate
+      if vehicle.plate_type == :ev
+       @collected_fees += 200
+      elsif vehicle.plate_type == :antique
+        @collected_fees += 25
+      else
+        @collected_fees += 100
+      end
+      return [vehicle]
+    else 
+      return false
+    end
+  end
```

Comment:
Good job with the logic here. Moving forward, we're going to want to start practicing SRP (Single Responsibility PRinciple) where each method is only responsible for doing one thing. This method is doing a lot. :) Creating more helper methods to move this logic into is how we solve that.

Reply:
That makes sense. I thought I was creating efficiency by putting them together, but I see now that it complicates the readability and functionality. I will use more helper methods in the future.